### ISSUE TITLE
Extra curly braces in redis key

### ISSUE THREAD

**Issue by Oscaner** [taylorhakes/python-redis-cache#27]
<img width="248" alt="image" src="https://github.com/taylorhakes/python-redis-cache/assets/26829325/205ef91e-8dfd-4d37-8af4-7f121348b3f8">

<img width="469" alt="image" src="https://github.com/taylorhakes/python-redis-cache/assets/26829325/14a247cc-df68-4ea4-95b2-ad704e4e20b8">

**Comment by morrissimo** [comment]
Seeing this behavior as well.

Here's a modified version of the demo script from the docs:
```python
from redis import StrictRedis
from redis_cache import RedisCache

client = StrictRedis(host="redis", decode_responses=True)
cache = RedisCache(redis_client=client)

print(f"{cache.prefix=}")
assert cache.prefix == "rc"


@cache.cache()
def my_func(arg1):
    print(f"[my_func] processing value: {arg1=}")
    return arg1


print(f"{client.keys('*')=}")
# should see no keys - eg:
# client.keys('*')=[]
assert len(client.keys("*")) == 0

print(f"{client.keys('rc:*')=}")
# filtering for key prefix, should still see no keys - eg:
# client.keys('rc:*')=[]
assert len(client.keys("rc:*")) == 0

# Use the function
my_func(1)
# should see print() output from my_func - eg:
# "[my_func] processing value: arg1=1"

print(f"{client.keys('*')=}")
# should see 1 key - eg:
# client.keys('*')=['rc:__main__:my_func:(some-b64-encoded-string)']
assert len(client.keys("*")) == 1

print(f"{client.keys('rc:*')=}")
# filtering for key prefix, should still see 1 key - eg:
# client.keys('*')=['rc:__main__:my_func:(some-b64-encoded-string)']
# ...but we don't, so this next assert will fail
assert len(client.keys("rc:*")) == 1

```
script output:
```
cache.prefix='rc'
client.keys('*')=[]
client.keys('rc:*')=[]
[my_func] processing value: arg1=1
client.keys('*')=['{rc:__main__.my_func}:eyJhcmcxIjoxfQ==']
client.keys('rc:*')=[]
Traceback (most recent call last):
  File "/app/wtf_redis_cache.py", line 41, in <module>
    assert len(client.keys("rc:*")) == 1
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

tested with
* python 3.11.7
* python-redis-cache 2.2.5

**Comment by morrissimo** [comment]
more context: change was introduced in commit https://github.com/taylorhakes/python-redis-cache/commit/69c13775aeec21e8994228d61e22482a0c12a324 and released in 2.1.0, including the test updates necessary to accommodate - so clearly not a bug. Perhaps all that's actually needed here is a few tweaks to the README, to discuss the generated key format/structure

**Comment by taylorhakes** [comment]
Hey, this was an accidental change as part of some experimentation. Sorry about that. Unfortunately changing the format back is backwards incompatible, so may need to change major version

**Comment by taylorhakes** [comment]
Fixed in version 3.0.0.

**Comment by taylorhakes** [comment]
Hey all, I couldn't for the life of me remember why I made the change to add curly braces (I really should have added a comment). I finally remembered when I got this report. 
https://github.com/taylorhakes/python-redis-cache/issues/29

Redis cluster requires all keys is to be in the same key space when operating in batches. Here is a link to someone explaining  on stack overflow
https://stackoverflow.com/a/72467085
https://semaphoreci.com/blog/redis-architectures#:~:text=If%20you%20operate%20on%20multiple,key%20that%20will%20be%20hashed.

For now I am going to roll this change back. We can evaluate other options if there are any.
